src/rocksdb.rs

```diff
@@ -513,6 +513,71 @@ impl DB {
         self.get_cf_opt(cf, key, &ReadOptions::new())
     }
 
+    pub fn multi_get_cf_opt(&self,
+                            cf: &CFHandle,
+                            keys: &[Vec<u8>],
+                            readopts: &ReadOptions)
+                            -> Result<Vec<Option<DBVector>>, String> {
+        unsafe {
+            let num_keys = keys.len() as size_t;
+            let mut keys_list = Vec::with_capacity(num_keys);
+            let mut keys_list_sizes = Vec::with_capacity(num_keys);
+            for i in 0..num_keys {
+                keys_list.push(keys[i].as_ptr());
+                keys_list_sizes.push(keys[i].len() as size_t);
+            }
+            let keys_list = keys_list.as_ptr();
+            let keys_list_sizes = keys_list_sizes.as_ptr();
+            let mut value_list = Vec::with_capacity(num_keys);
+            let mut value_list_sizes = Vec::with_capacity(num_keys);
+            let mut err_list = Vec::with_capacity(num_keys);
+            crocksdb_ffi::crocksdb_multi_get_cf(self.inner,
+                                                readopts.get_inner(),
+                                                cf.inner,
+                                                num_keys,
+                                                keys_list,
+                                                keys_list_sizes,
+                                                value_list.as_mut_ptr(),
+                                                value_list_sizes.as_mut_ptr(),
+                                                err_list.as_mut_ptr());
+            let mut value_vec = Vec::with_capacity(num_keys);
+            for i in 0..num_keys {
+                if !err_list[i].is_null() {
+                    return Err(crocksdb_ffi::error_message(err_list[i]));
```

Comment:
There are still memory leaks. But the content in value_list and err_list should be dropped.